students/eric_grandeo/lesson03/list_lab.py

```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env python3
+
+#Series 1
+fruits = ["Apples", "Pears", "Oranges", "Peaches"]
+print(fruits)
+
+#Ask the user for another fruit and add it to the end of the list.
+response = input("Please enter in a new fruit > ")
+fruits.append(response)
+
+print(fruits)
+
+#Ask the user for a number and display the number back to the user
+#and the fruit corresponding to that number (on a 1-is-first basis)
+num = int(input("Please enter a number > "))
+print("You selected number {}, which is {}".format(num, fruits[num-1]))
+
+#Add another fruit to the beginning of the list using “+” and display the list.
+strawberry = ['Strawberry']
+fruits = strawberry + fruits
+
+print("Added new fruit with +: {}".format(fruits))
+
+#Add another fruit to the beginning of the
+#list using insert() and display the list.
+
+fruits.insert(0, 'Banana')
+print("Added new fruit with insert: {}".format(fruits))
+
+#Display all the fruits that begin with “P”, using a for loop.
+letter_p = []
+for i in fruits:
+    if i[0] == "P":
+        letter_p.append(i)
+
+print("Fruits that begin with 'P': {}".format(letter_p))
+
+fruits_series_one = fruits[:]
+
+#Series 2
+print("Series 2{}".format('-' * 25))
+print("Here is the fruits list: {}".format(fruits))
+
+#Remove the last fruit from the list.
+fruits.pop(-1)
+print("Removed the last fruit: {}".format(fruits))
+
+#Ask the user for a fruit to delete, find it and delete it.
+del_fruit = input("Please enter a fruit you would like to delete > ")
+fruits.remove(del_fruit)
+print("Removed the fruit per your request: {}".format(fruits))
+
+#Series 3
+print("Series 3{}".format('-' * 25))
+print("Here is the fruits list from Series 1: {}".format(fruits_series_one))
+
+#Ask the user for input displaying a line like “Do you like apples?”
+#for each fruit in the list (making the fruit all lowercase).
+new_fruits = []
+
+for x in fruits_series_one:
+    keep_going = True
+    while keep_going:
+        user_like = input("Do you like {} ? ".format(x.lower()))
+        if user_like.lower() == "yes":
+            new_fruits.append(x)
+            #I didnt delete from original list; deleting in place caused errors
+            keep_going = False
```

Comment:
Thank you for the help. In this example, do you put the if statements for yes (do nothing) and no (fruits.remove(x)) in the else clause?

Reply:
this is a complete example:
```
for fruit in fruits[:]:
    while True:
        user_response = input(f"Do you like {fruit.lower()} ? ")
        if user_response not in ("yes", "no"):
            print("Please answer yes or no")
        else:
            break
        
    if user_response == "yes":
        new_fruits.append(fruit)
```

think of your while True loop just responsible for getting correct user response, for example it could also look like this:
```
def get_valid_user_response():
    while True:
        user_response = input(f"Do you like {fruit.lower()} ? ")
        if user_response not in ("yes", "no"):
            print("Please answer yes or no")
        else:
            return user_response


for fruit in fruits[:]:
    if get_valid_user_response() == "yes":
        new_fruits.append(fruit)
```